hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/common/utils/BufferUtils.java

```diff
@@ -136,4 +139,19 @@ public static int getNumberOfBins(long numElements, int maxElementsPerBin) {
     }
     return Math.toIntExact(n);
   }
+
+  /**
+   * Write all remaining bytes in buffer to the given channel.
+   */
+  public static long writeFully(GatheringByteChannel ch, ByteBuffer bb) throws IOException {
+    long written = 0;
+    while (bb.remaining() > 0) {
+      int n = ch.write(bb);
+      if (n <= 0) {
```

Comment:
According to the javadoc, `n == 0` is a valid case.

BTW, let's have another method to handle array?  The `GatheringByteChannel` works more efficient in that way.
```java
  public static long writeFully(GatheringByteChannel ch, ByteBuffer[] buffers) throws IOException {
    long written = 0;
    for(int i = 0; i < buffers.length; i++) {
      while (buffers[i].remaining() > 0) {
        final long n = ch.write(buffers, i, buffers.length - i);
        if (n < 0) {
          throw new IllegalStateException("GatheringByteChannel.write returns " + n + " for " + ch);
        }
        written += n;
      }
    }
    return written;
  }
```

Reply:
Thanks @szetszwo for the suggestion.  Can we address it in follow-up?

Reply:
Sure, let's do it in a follow up.

Reply:
How about merging this first?  We can change it immediately in the follow-up pr.

Reply:
HDDS-11860